packages/core/src/search/orama-cloud.ts

```diff
@@ -84,27 +89,41 @@ export interface OramaIndex {
 }
 
 export async function sync(
-  cloudManager: CloudManager,
+  cloudManager: OramaCloud,
   options: SyncOptions,
 ): Promise<void> {
   const { autoDeploy = true } = options;
-  const index = cloudManager.index(options.index);
+  const index = cloudManager.index.set(options.index);
+
+  // Open a new Orama transaction.
+  // This will create a hidden, temporary empty index we can push new documents to.
+  await index.transaction.open();
 
-  await index.snapshot(options.documents.flatMap(toIndex));
-  if (autoDeploy) await index.deploy();
+  // Insert the documents into the temporary index.
+  await index.transaction.insertDocuments(
```

Comment:
The current `index.snapshot` call may fail when there are too many documents and the payload is too large. Will the new `insertDocuments` call have the same issue?

Thanks.